library/core/src/ptr/mod.rs

```diff
@@ -430,6 +430,10 @@ mod mut_ptr;
 /// done automatically by the compiler. This means the fields of packed structs
 /// are not dropped in-place.
 ///
+/// [`drop_in_place()`] does not modify the pointed-to value beyond any changes
+/// performed by [`Drop::drop()`]. As far as the compiler is concerned, the value
+/// will still contain a valid bit pattern for type `T`.
```

Comment:
```suggestion
/// must still contain a valid bit pattern for type `T`.
```

I feel like "will" is a little odd here, this reads better to me -- but in general I find the sentence strange; typically we don't I think say things like "As far as the compiler ...", rather that something is UB or not.

Reply:
You'd say that except that party of the wording was lifted straight from the MaybeUninit docs :)

As Jakob says it doesn't make much sense though, so we should probably tighten up the wording on both

Reply:
You mean the ManuallyDrop docs, I assume. But yeah that wording doesn't make a ton of  sense. I don't know what that sentence is even trying to achieve, can we just remove it?